pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,288 @@
+				ClusterID:                "",
+				InfraID:                  "",
+				AdminKubeconfigSecretRef: v1.LocalObjectReference{Name: fmt.Sprintf("%s-admin-kubeconfig", i.localClusterNamespace)},
+			},
+			ClusterInstallRef: &hivev1.ClusterInstallLocalReference{
+				Name:    i.localClusterNamespace + "-cluster-install",
+				Group:   "extensions.hive.openshift.io",
+				Kind:    "AgentClusterInstall",
+				Version: "v1beta1",
+			},
+			Platform: hivev1.Platform{
+				AgentBareMetal: &agent.BareMetalPlatform{
+					AgentSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"infraenv": "local-cluster"},
+					},
+				},
+			},
+			PullSecretRef: &v1.LocalObjectReference{
+				Name: pullSecret.Name,
+			},
+		},
+	}
+	clusterDeployment.Name = i.localClusterNamespace + "-cluster-deployment"
+	clusterDeployment.Namespace = i.localClusterNamespace
+	clusterDeployment.Spec.ClusterName = i.localClusterNamespace + "-cluster-deployment"
+	clusterDeployment.Spec.BaseDomain = dns.Spec.BaseDomain
+	//
+	// Adding this ownership reference to ensure we can submit clusterDeployment without ManagedClusterSet/join permission
+	//
+	// Must match https://github.com/stolostron/multicloud-operators-foundation/blob/0001f46a5115fe43c606a068e5e7ee00abec3b68/pkg/webhook/clusterset/validatingWebhook.go#L44
+	// or the ClusterDeployment will be rejected during admission.
+	agentClusterInstallOwnerRef := metav1.OwnerReference{
+		Kind:       "AgentCluster",
+		APIVersion: "capi-provider.agent-install.openshift.io/v1alpha1",
+		Name:       i.localClusterNamespace + "-cluster-install",
+		UID:        agentClusterInstall.UID,
+	}
+	clusterDeployment.OwnerReferences = []metav1.OwnerReference{agentClusterInstallOwnerRef}
+	_, err := i.clusterImportOperations.CreateClusterDeployment(clusterDeployment)
+	if err != nil {
+		i.log.Errorf("could not create ClusterDeployment due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (i *LocalClusterImport) shouldImportLocalCluster() (bool, error) {
+	// The presence of agentClusterInstall for the local cluster will determine whether or not we need to perform an import
+	agentClusterInstall, err := i.clusterImportOperations.GetAgentClusterInstall(i.localClusterNamespace, i.localClusterNamespace+"-cluster-install")
+	if err != nil && !k8serrors.IsNotFound(err) {
+		i.log.Errorf("there was an error fetching the AgentClusterInstall while checking for need to register local cluster.")
+		return false, err
+	}
+	// TODO: We need to determine what to do if there has been a recent upgrade
+	// See MGMT-15705 for more information.
+	if agentClusterInstall != nil {
+		i.log.Infof("found AgentClusterInstall for hub cluster, assuming that hub has been correctly registered for assisted day 2 operations: %s", agentClusterInstall.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (i *LocalClusterImport) ImportLocalCluster() error {
+
+	shouldImportLocalCluster, err := i.shouldImportLocalCluster()
```

Comment:
As we talked, this function can be removed and you will try to create all the objects and if they are already exist you will ignore them, using `k8serrors.IsAlreadyExists(err)` to check the error